plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/transition/MapboxViewportTransitionFactory.kt

```diff
@@ -0,0 +1,278 @@
+package com.mapbox.maps.plugin.viewport.transition
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.view.animation.Interpolator
+import androidx.core.view.animation.PathInterpolatorCompat
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.animation.CameraAnimatorOptions.Companion.cameraAnimatorOptions
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.viewport.ViewportPluginImpl.Companion.VIEWPORT_CAMERA_OWNER
+import kotlin.math.abs
+
+/**
+ * Helper class that provides default implementation of [ViewportTransition] generators.
+ */
+internal class MapboxViewportTransitionFactory(
+  mapDelegateProvider: MapDelegateProvider
+) {
+  private val cameraManager = mapDelegateProvider.mapCameraManagerDelegate
+  private val cameraPlugin = mapDelegateProvider.mapPluginProviderDelegate.camera
+
+  fun transitionFromLowZoomToHighZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
+    val currentMapCameraState = cameraManager.cameraState
+
+    var centerDuration = 0L
+    cameraOptions.center?.let { center ->
+      val screenDistanceFromMapCenterToLocation =
+        screenDistanceFromMapCenterToTarget(
+          cameraManager = cameraManager,
+          currentCenter = currentMapCameraState.center,
+          targetCenter = center
+        )
+
+      val centerAnimationRate = 500L
```

Comment:
can we move this (and all similar vals below in this file) to const? why do we need to declare those vals each time

Reply:
these factory utilities are ported from nav sdk, although I see the point of keeping them out of constants, as these are very specific value corresponding to specific situations/functions. And most of these values are not reused by other animators, so I'd prefer to leave it as is.